library/std/src/primitive_docs.rs

```diff
@@ -1097,23 +1099,56 @@ mod prim_ref {}
 /// let really_safe_ptr: unsafe fn(usize) -> usize = add_one;
 /// ```
 ///
-/// On top of that, function pointers can vary based on what ABI they use. This is achieved by
-/// adding the `extern` keyword to the type name, followed by the ABI in question. For example,
-/// `fn()` is different from `extern "C" fn()`, which itself is different from `extern "stdcall"
-/// fn()`, and so on for the various ABIs that Rust supports. Non-`extern` functions have an ABI
-/// of `"Rust"`, and `extern` functions without an explicit ABI have an ABI of `"C"`. For more
-/// information, see [the nomicon's section on foreign calling conventions][nomicon-abi].
+/// ### ABI
+///
+/// On top of that, function pointers can vary based on what ABI they use. This
+/// is achieved by adding the `extern` keyword before the type, followed by the
+/// ABI in question. The default ABI is "Rust", i.e., `fn()` is the exact same
+/// type as `extern "Rust" fn()`. A pointer to a function with C ABI would have
+/// type `extern "C" fn()`.
+///
+/// `extern "ABI" { ... }` blocks declare functions with ABI "ABI". The default
+/// here is "C", i.e., functions declared in an `extern {...}` block have "C"
+/// ABI.
+///
+/// For more information and a list of supported ABIs, see [the nomicon's
+/// section on foreign calling conventions][nomicon-abi].
 ///
-/// [nomicon-abi]: ../nomicon/ffi.html#foreign-calling-conventions
+/// ### Variadic functions
 ///
 /// Extern function declarations with the "C" or "cdecl" ABIs can also be *variadic*, allowing them
-/// to be called with a variable number of arguments. Normal rust functions, even those with an
+/// to be called with a variable number of arguments. Normal Rust functions, even those with an
 /// `extern "ABI"`, cannot be variadic. For more information, see [the nomicon's section on
 /// variadic functions][nomicon-variadic].
 ///
 /// [nomicon-variadic]: ../nomicon/ffi.html#variadic-functions
 ///
-/// These markers can be combined, so `unsafe extern "stdcall" fn()` is a valid type.
+/// ### Creating function pointers
+///
+/// When `bar` is the name of a function, then the expression `bar` is *not* a
+/// function pointer. Rather, it denotes a value of an unnameable zero-sized
+/// type that uniquely identifies the function `bar`. This type *coerces* to a
```

Comment:
Can we mention that function item value doesn't store actual function pointer?

Reply:
What do you mean? It is a zero-sized type; it cannot store anything?

Reply:
I mean to make it clear and to confirm that function item doesn't contain function pointer.
@LukasKalbertodt has a related answer on stackoverflow: https://stackoverflow.com/a/54629954/5456794

Reply:
I think that it is fine as-is, but you are of course welcome to submit follow-up PRs with specific language.

Reply:
I like some of the wording in that post and will try to incorporate it.

Reply:
@RalfJung In case with "that post" you are talking about my StackOverflow answer, note that it is licensed as CC0 (which I also state on [my profile](https://stackoverflow.com/users/2408867/lukas-kalbertodt?tab=profile)). So feel free to copy whatever you want. Just FYI.

Reply:
@LukasKalbertodt that's great, thanks! I am not sure if individual sentences are copyrighted but this means I do not have to worry. :D

Reply:
Done; @Mark-Simulacrum please check.